### PR TITLE
Conditional rendering of follow button

### LINKED ISSUE #115: Follow/Unfollow Button
button to follow/unfollow people in author page

### PR DESCRIPTION
**Pull request by derrick-w**
Follow/unfollow button will:
- Not render if profile being viewed is the logged in user's profile
- Render follow button if viewing profile of a user the logged in user does not currently follow
- Render unfollow button if viewing profile of a user the logged in user currently follows

Resolves #115 
image

### CODE REVIEW

File: frontend/src/pages/Author.jsx [hunk 1]
```diff
@@ -56,8 +67,169 @@ class Author extends Component {
...
+        let urlPath = "/api/unfollow/"
+		let body = {
+			query: "unfollow",
+			author: {
+				id: authorID[0],
+				host: authorID[1],
+			},
+			friend:{
+				id: this.props.location.state.fullAuthorId,
+				host: this.props.location.state.host,
```
Comment (hiufungk) [inline frontend/src/pages/Author.jsx]: this.props.location.state.host I don't think this exist at all you have to pass it into your helper function to get it

File: frontend/src/pages/Author.jsx [hunk 2]
```diff
@@ -56,8 +68,169 @@ class Author extends Component {
...
+                    httpResponse.json().then((results) => {
+                        this.setState({
+                            isFollowing: false
+                        })
+						toast(
+							{
+								type: 'success',
+								icon: 'user',
+								title: 'Unfollow successful!',
+								description: <p>You are now following {this.props.location.state.displayName}</p>
```
Comment (hiufungk) [inline frontend/src/pages/Author.jsx]: this.props.location.state.displayNameI don't think this exist at all, you have to get it from this.state

**Review by hiufungk — CHANGES_REQUESTED** [review]

File: frontend/src/pages/Author.jsx [hunk 3]
```diff
@@ -37,6 +46,8 @@ class Author extends Component {
             .then((httpResponse) => {
                 if (httpResponse.status === 200) {
                     httpResponse.json().then((results) => {
+                        var authorID;
```
Comment (htruong4) [inline frontend/src/pages/Author.jsx]: You can do this in one line.
const authorID = this.getloggedinAuthorIDandHost()

File: frontend/src/pages/Author.jsx [hunk 4]
```diff
@@ -56,8 +68,169 @@ class Author extends Component {
...
+        if (cookiehost !== null) {
+            host = cookiehost;
+        } else if (storehost !== null) {
+            host = storehost;
+        }
+        return [authorID, host]
+    }
+
+    getFollowStatus() {
+        var authorID;
```
Comment (htruong4) [inline frontend/src/pages/Author.jsx]: You can do this in one line as above

File: frontend/src/pages/Author.jsx [hunk 5]
```diff
@@ -56,8 +68,169 @@ class Author extends Component {
...
+        authorID = this.getloggedinAuthorIDandHost()
+        let urlPath = "/api/followers/" + utils.getShortAuthorId(this.props.location.state.fullAuthorId),
+            requireAuth = true;
+        HTTPFetchUtil.getRequest(urlPath, requireAuth)
+            .then((httpResponse) => {
+                if (httpResponse.status === 200) {
+                    httpResponse.json().then((results) => {
+                        for (var i = 0; i < results.authors.length; i++) {
```
Comment (htruong4) [inline frontend/src/pages/Author.jsx]: use let because it is only valid within the scope of the for loop.

File: frontend/src/pages/Author.jsx [hunk 6]
```diff
@@ -56,8 +68,169 @@ class Author extends Component {
...
+                    })
+                }
+            })
+            .catch((error) => {
+                console.error(error);
+        });    
+        } 
+
+    sendUnfollowRequest() {
+        var authorID;
```
Comment (htruong4) [inline frontend/src/pages/Author.jsx]: const as above

File: frontend/src/pages/Author.jsx [hunk 7]
```diff
@@ -56,8 +68,169 @@ class Author extends Component {
...
+								icon: 'user',
+								title: 'Unfollow successful!',
+								description: <p>You are now following {this.props.location.state.displayName}</p>
+							}
+						);
+
+                    })
+				}
+				else{
+					console.log(httpResponse)
```
Comment (htruong4) [inline frontend/src/pages/Author.jsx]: If you don't need this console log, would you be able to remove this?

File: frontend/src/pages/Author.jsx [hunk 8]
```diff
@@ -56,8 +68,169 @@ class Author extends Component {
...
+								icon: 'user',
+								title: 'Follow successful!',
+								description: <p>You are now following {this.props.location.state.displayName}</p>
+							}
+						);
+
+                    })
+				}
+				else{
+					console.log(httpResponse)
```
Comment (htruong4) [inline frontend/src/pages/Author.jsx]: remove the console.log if you don't need this anymore

**Review by htruong4 — CHANGES_REQUESTED** [review]

File: frontend/src/pages/Author.jsx (hunk 1 above)
Comment (derrick-w) [reply frontend/src/pages/Author.jsx]: fixed

File: frontend/src/pages/Author.jsx (hunk 2 above)
Comment (derrick-w) [reply frontend/src/pages/Author.jsx]: fixed

File: frontend/src/pages/Author.jsx (hunk 7 above)
Comment (derrick-w) [reply frontend/src/pages/Author.jsx]: ok

File: frontend/src/pages/Author.jsx (hunk 8 above)
Comment (derrick-w) [reply frontend/src/pages/Author.jsx]: ok

File: frontend/src/pages/Author.jsx (hunk 3 above)
Comment (derrick-w) [reply frontend/src/pages/Author.jsx]: ok

File: frontend/src/pages/Author.jsx (hunk 4 above)
Comment (derrick-w) [reply frontend/src/pages/Author.jsx]: ok

File: frontend/src/pages/Author.jsx (hunk 5 above)
Comment (derrick-w) [reply frontend/src/pages/Author.jsx]: ok

File: frontend/src/pages/Author.jsx (hunk 6 above)
Comment (derrick-w) [reply frontend/src/pages/Author.jsx]: ok

**Review by htruong4 — APPROVED** [review]

File: frontend/src/pages/Author.jsx [hunk 9]
```diff
@@ -56,8 +67,166 @@ class Author extends Component {
...
+                    httpResponse.json().then((results) => {
+                        for (let i = 0; i < results.authors.length; i++) {
+                            if (results.authors[i].id === authorID[0]){
+                                this.setState({
+                                    isFollowing: true
+                                })
+                            }
+                        }
+                    })
+                }
```
Comment (shchau) [inline frontend/src/pages/Author.jsx:146]: Could add a toasty boi here for non-200 responses. 
But I approve anyways since everything else looks fine 👍

**Review by shchau — APPROVED** [review]

**Review by hiufungk — APPROVED** [review]

**Pull request merged**